src/backend/distributed/commands/create_citus_local_table.c

```diff
@@ -494,26 +496,63 @@ GetRenameShardIndexCommand(char *indexName, uint64 shardId)
 
 
 /*
- * RenameShardRelationTriggers appends given shardId to the end of the names
- * of shard relation triggers that are explicitly created. This function
- * utilizes GetExplicitTriggerNameList to pick the triggers to be renamed, see
- * more details in function's comment.
+ * RenameOrDropShardRelationTriggers appends given shardId to the end of the
+ * names of shard relation INSERT/DELETE/UPDATE triggers and drops TRUNCATE
+ * triggers that are explicitly created. This function utilizes
+ * GetExplicitTriggerNameList to pick the triggers to be renamed, see more
+ * details in function's comment.
  */
 static void
-RenameShardRelationTriggers(Oid shardRelationId, uint64 shardId)
+RenameOrDropShardRelationTriggers(Oid shardRelationId, uint64 shardId)
```

Comment:
I think it's shard relation at that time. This is because we rename the relation in `RenameRelationToShardRelation` and then do such index, trigger etc. rename things

Reply:
Yeah that seems right. 
I guess it seems a bit weird that we are appending shardID to shardRelation.

When we convert local table to a shard relation, not sure if we would say that the constraints etc of local table now belongs to the shard table. Or like they belong to the shard relation only after the renaming is done.

An alternative would look like this:

```c
static uint64
ConvertLocalTableToShard(Oid relationId)
{
	uint64 shardId = GetNextShardId();

	RenameRelationToShardRelation(relationId, shardId);
	RenameLocalRelationConstraints(relationId, shardId);
	RenameLocalRelationIndexes(relationId, shardId);
	DropLocalRelationTruncateTriggers(relationId);
	RenameLocalRelationNonTruncateTriggers(relationId, shardId);

	return shardId;
}
```

I guess this is kind of a minor thing, feel free to ignore this.